### ISSUE TITLE
Custom decoder and alert test fine but don't show on front end or send alerts

### ISSUE THREAD

**Issue by TheLeprech4un** [wazuh/wazuh#21604]
Hi All,

I need some help.

My custom decoder and alert show they are working fine when using /var/ossec/bin/wazuh-logtest

However, when I perform a real life test, the alerts are not sent and I can't see any evidence of it on the Wazuh portal.

Here is my custom decoder setup:

```
<decoder name="watchguard">
    <prematch>^Member\d</prematch>
</decoder>

<decoder name="watchguard_login">
    <parent>watchguard</parent>
    <regex offset="after_parent">^->(\d+.\d+.\d+.\d+) (\w\w\w \d\d \d\d:\d\d:\d\d) Member\d (\w+) (\w+-\w+-\w+) \p\w\w\w\w-\w\w-\w\w\w\w\w:\w\w:\w\w\p sessiond\p(\d+)\p: msg_id=\p\w+-\w+\p Management user (\w+)@Firebox-DB from (\d+.\d+.\d+.\d+) logged in</regex>
    <order>fw_ip,timestamp,fw_serial,fw_model,session_id,user_login,source_ip</order>
</decoder>
```

Here is my rule:

```
<group name="watchguard,syslog">
     <rule id="100005" level="7">
         <decoded_as>watchguard</decoded_as>
         <field name="user_login">admin-shale</field>
         <description>Watchguard login detected: $(user_login) from $(source_ip)</description>
         <options>alert_by_email</options>
     </rule>
</group>
```

Here is the output from /var/ossec/bin/wazuh-logtest

```
**Phase 1: Completed pre-decoding.
        full event: 'Member2->192.168.0.*** Jan 24 11:57:49 Member2 ********* CMS-T85-1 (2024-01-24T11:57:49) sessiond[2741]: msg_id="3E00-***" Management user admin-shale@Firebox-DB from 192.168.0.*** logged in'

**Phase 2: Completed decoding.
        name: 'watchguard'
        fw_ip: '192.168.0.***'
        fw_model: 'CMS-T85-1'
        fw_serial: '*********'
        session_id: '2741'
        source_ip: '192.168.0.***'
        timestamp: 'Jan 24 11:57:49'
        user_login: 'admin-shale'

**Phase 3: Completed filtering (rules).
        id: '100005'
        level: '7'
        description: 'Watchguard login detected: admin-shale from 192.168.0.***'
        groups: '['watchguard', 'syslog']'
        firedtimes: '1'
        mail: 'True'
**Alert to be generated.
```

(I have redacted some info)

The above should generate an alert whenever user admin-shale logs in. However, this is not happening.

I have enabled log_all and monitored the archive.log and I can see the syslogs are coming through. However, Wazuh does nothing with them. 

Using version 4.7.2.

Please help I am running low on ideas. Everything suggests it should be working but I cannot see any events for it in the portal, and no alerts are generated.

**Comment by TheLeprech4un** [comment]
FYI, I thought it could be something to do with the below:

```
<!--
  - Allowed static fields:
  - location   - where the log came from (only on FTS)
  - srcuser    - extracts the source username
  - dstuser    - extracts the destination (target) username
  - user       - an alias to dstuser (only one of the two can be used)
  - srcip      - source ip
  - dstip      - dst ip
  - srcport    - source port
  - dstport    - destination port
  - protocol   - protocol
  - id         - event id
  - url        - url of the event
  - action     - event action (deny, drop, accept, etc)
  - status     - event status (success, failure, etc)
  - extra_data - Any extra data
-->
```

So I changed my decoder to:

<decoder name="watchguard">
    <prematch>^Member\d</prematch>
</decoder>

```
<decoder name="watchguard_login">
    <parent>watchguard</parent>
    <regex offset="after_parent">^->(\d+.\d+.\d+.\d+) \w\w\w \d\d \d\d:\d\d:\d\d Member\d \w+ \w+-\w+-\w+ \p\w\w\w\w-\w\w-\w\w\w\w\w:\w\w:\w\w\p sessiond\p(\d+)\p: msg_id=\p\w+-\w+\p Management user (\w+)@Firebox-DB from (\d+.\d+.\d+.\d+) logged in</regex>
    <order>dstip,id,user,srcip</order>
</decoder>
```

However this made no difference.

**Comment by Sinistr0** [comment]
I have exactly the same problem, the decoder and the rule work in logtest, the logs appear in archive.log but the rules do not appear in alerts.json or in the dashboard

**Comment by TheLeprech4un** [comment]
I think it has something to do with including "Member2->" part of the log. As the raw syslog data does not contain this. However, removing that is causing me even more problems.

If I change the prematch data to the below:

```
<decoder name="watchguard">
    <prematch>^\w\w\w \d\d \d\d:\d\d:\d\d</prematch>
</decoder>
```

Nothing I put in the child decoder works.

```
<decoder name="watchguard_login">
    <parent>watchguard</parent>
    <regex offset="after_parent">Member\d \w+ \w+-\w+-\w+ \p\w\w\w\w-\w\w-\w\w\w\w\w:\w\w:\w\w\p sessiond\p(\d+)\p: msg_id=\p\w+-\w+\p Management user (\w+)@Firebox-DB from (\d+.\d+.\d+.\d+) logged in</regex>
    <order>id,user,srcip</order>
</decoder>
```

The above should match but it doesn't. Here is the raw syslog data (redacted):

```
Jan 24 14:09:42 Member2 *********** CMS-T85-1 (2024-01-24T14:09:42) sessiond[2741]: msg_id="3E00-0002" Management user admin-shale@Firebox-DB from 192.168.0.*** logged in
```

I don't understand why the above doesn't match

**Comment by TheLeprech4un** [comment]
I've managed to solve the problem.

I've just read this in the documentation:

 _if the log is Syslog-like, then prematch only analyzes the log after the Syslog-like header. If the log is not Syslog-like, then it analyzes the entire log._

Taken from here: [](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/decoders.html#decoders-prematch)

So, initially I was running my regex expression starting from the beginning of the log, as below:

```
Member2->192.168.0.* Jan 24 13:38:58 Member2 C03*********
```

However, I then realised the _Member2 -> 192.168.0.250_ is not part of the syslog and only displays in Wazuh's archive.log file.

So, I then tried starting the regex from the datestamp, as below:

```
Jan 24 13:38:58 Member2 C03*********
```

However, Wazuh assumes the datestamp and _Member2_ are syslog headers and ignores them. It was classing Member2 as the hostname.

Now I have modified my expression to start from the serial number _C03*********_ it works correctly. Entries are now displayed in the web portal and alerts are sent.

Hope this helps anyone with the same problem.
